go.mod

```diff
@@ -113,3 +113,7 @@ require (
 )
 
 go 1.20
+
+replace github.com/codeready-toolchain/api => github.com/filariow/toolchain-api v0.0.0-20240621164613-21096009ec46
+
+replace github.com/codeready-toolchain/toolchain-common => github.com/filariow/toolchain-common v0.0.0-20240621225435-5fc3b6ce2cc2
```

Comment:
Let's update the api dependency later (as part of your implementation changes) - for smooth PR merge process of this PR.
You added only a new API and new functions, so there is nothing it could break in the current logic.

Reply:
Just to be sure I've understood correctly, do you think we can close this PR in favor of #1047 ?

Reply:
oh, got it! Thanks for the explanation.